### PR TITLE
Remove boost::any dependency from xrt_device.h

### PR DESCRIPTION
**Pull request by stsoe**
#### Problem solved by the commit
Default building of XRT to use std=c++17 and use std::any in place of
boost::any in the xrt::device exposed get_info() API.  This removes
the need to install boost-dev when using native XRT APIs.

#### How problem was solved, alternative solutions (if any) and why they were rejected
Support backwards compatibility with host code compiled with earlier
versions of XRT.  If host application was compiled against earlier
version of XRT it references boost::any get_info symbols.  These
symbols are retained and old binaries will continue to work with new
XRT.  Also new binaries that are built with -std=c++14 will use
boost::any and require that boost-dev is installed.

The boost::any version of get_info() is hereby deprecated and will
be removed in the future per deprecation policy.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Installing XRT on a bare system doesn't install boost-dev package, 
hence using the native APIs (xrt::device) didn't work out of the box.
This problem was introduced in xrt-2.11.x where xrt::device::get_info
was added.

#### Risks (if any) associated the changes in the commit
Switching to -std=c++17 implies we no longer support some older
compilers,.  In particular Ubuntu-16.04 can no longer be used to
build XRT without specifying `./build.sh -cppstd 14 ...`

#### What has been tested and how, request additional testing if necessary
Backwards compatibility has been checked with tests/xrt/query/  where an
earlier built binary has been verified to work with latest xrt.

#### Documentation impact (if any)
@uday610 we should document that c++17 capable compiler is required 
when building XRT.  For now user space code can still build using c++14, but 
will  need to manually install boost-dev.

This PR resolves #6111 which can be closed after this PR is merged.

### CODE REVIEW

**Review by jvillarre — APPROVED** [review]

File: src/CMake/nativeWin.cmake [hunk 1]
```diff
@@ -33,7 +33,17 @@ find_package(Boost
 set(XRT_BOOST_VERSION ${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}.${Boost_SUBMINOR_VERSION})
```
Comment (rozumx) [inline src/CMake/nativeWin.cmake:33]: Line 2 copyright date needs to be updated.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -19,7 +19,9 @@ include_directories(
   ${CMAKE_BINARY_DIR}/gen
```
Comment (rozumx) [inline src/CMakeLists.txt:19]: Line 2 copyright date needs to be updated.

File: src/runtime_src/core/common/api/xrt_device.cpp [hunk 3]
```diff
@@ -79,21 +79,21 @@ namespace query {
 // that the query request result type matches the xrt::info::device
```
Comment (rozumx) [inline src/runtime_src/core/common/api/xrt_device.cpp:79]: Line 2 copyright date needs to be updated.

File: src/runtime_src/core/include/xrt/CMakeLists.txt [hunk 4]
```diff
@@ -14,7 +14,8 @@ install (FILES ${XRT_XRT_HEADER_SRC} DESTINATION ${XRT_INSTALL_INCLUDE_DIR}/xrt
```
Comment (rozumx) [inline src/runtime_src/core/include/xrt/CMakeLists.txt:14]: Line 2 copyright date needs to be updated.

File: src/runtime_src/core/include/xrt/detail/any.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2021 Xilinx, Inc. All rights reserved.
```
Comment (rozumx) [inline src/runtime_src/core/include/xrt/detail/any.h:2]: Copyright date needs to be updated.

File: src/runtime_src/core/include/xrt/xrt_device.h [hunk 6]
```diff
@@ -24,16 +24,16 @@
```
Comment (rozumx) [inline src/runtime_src/core/include/xrt/xrt_device.h:24]: Line 2 copyright date needs to be updated.

File: tests/validate/CMakeLists.txt [hunk 7]
```diff
@@ -3,7 +3,9 @@
 #
```
Comment (rozumx) [inline tests/validate/CMakeLists.txt:3]: Line 2 copyright date needs to be updated.

File: tests/xrt/CMakeLists.txt [hunk 8]
```diff
@@ -3,8 +3,8 @@
 #
```
Comment (rozumx) [inline tests/xrt/CMakeLists.txt:3]: Line 2 copyright date needs to be updated.

**Review by rozumx — APPROVED** [review]
Approved.

**Review by uday610 — APPROVED** [review]

**Comment by dayeh-xilinx** [comment]
retest this please. Cancelled from queue to give way to PR#6159

**Review by chvamshi-xilinx — APPROVED** [review]

**Pull request merged**